core/src/mender-client.c

```diff
@@ -905,10 +905,16 @@ mender_client_update_work_function(void) {
                     goto END;
                 }
 
-                mender_log_info("Downloading deployment artifact with id '%s', artifact name '%s' and uri '%s'",
-                                deployment->id,
-                                deployment->artifact_name,
-                                deployment->uri);
+#if CONFIG_MENDER_LOG_LEVEL >= MENDER_LOG_LEVEL_INF
+                char id_short[7];
+                if (strlen(deployment->id) >= 7) {
+                    strncpy(id_short, deployment->id, 7);
+                    strcat(id_short, "...");
+                } else {
+                    strncpy(id_short, deployment->id, strlen(deployment->id));
+                }
+                mender_log_info("Downloading artifact with id '%s', name '%s', uri '%s'", id_short, deployment->artifact_name, deployment->uri);
```

Comment:
It doesn't work properly, I got for example an output like `artifact with id 'a8cf648 �%...', name`. I'll look at it at a later time of maybe you C experts can spot the bug.

Reply:
yup, we need space for 10 characters and then use 7 from the ID + "..." or all 10 from the ID if it fits.

Reply:
Btw, `%.7s` also works to limit the number of characters printed. So it might be easiest to do `if (strlen() <= 10) log(everything); else log(%.7s and with_dots)`.

Reply:
> Btw, %.7s also works to limit the number of characters printed

TIL! Thanks. I'll try this for a simpler to read code :clap: